src/validation.cpp

```diff
@@ -2130,13 +2134,12 @@ bool CChainState::ConnectBlock(const CBlock& block, BlockValidationState& state,
             return state.Invalid(BlockValidationResult::BLOCK_CONSENSUS, "bad-blk-sigops");
         }
 
-        txdata.emplace_back(tx);
```

Comment:
This line initializes txdata[i]. In the new code that is initialized... when exactly? somewhere inside `CheckInputScripts()`?

Reply:
In the new code the object is (dummy) initialized in the std::vector constructor above (in the C++ language definition of objects being constructed). Filling those objects with actual data now happens in CheckInputScripts when it calls the Init member function.

Reply:
oh, of course -- I misread that Init() call as being in a different function. thanks!